go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -619,6 +619,11 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 		}
 	}
 
+	// It is possible that IPs have already been allocated for this pod and annotation has been updated, then the last
+	// addLogicalPortToNetwork() failed afterwards. In the current retry attempt, if the input pod argument got from
+	// the informer cache still lags behind, we would fail to get the updated pod annotation. Just continue to allocate
```

Comment:
wow! nice catch! out of curiosity, how often was this observed? do you have some ovnk-master logs where the cache misbehaved in this manner?

Just curious because let's say in update1 we assigned podIPA to podA and updated the annotation to reflect that change, then let's say we failed to create LSP ops and addLogicalPortToNetwork returned with error, the next update2 of this podA I'd hope comes with that updated podIPA annotation...:/ was that not the case because of kapi<->informer lag was that predominant?

Reply:
yes, we ran into this because the subsequent lsp creation failed. I didn't save the ovnkube-master log though. But we only notice this type of error in that particular occation.